test/reporters/tests/tree.test.ts

```diff
@@ -0,0 +1,141 @@
+import { describe, expect, it } from 'vitest'
+import { runVitest, StableTestFileOrderSorter } from '../../test-utils'
+import { trimReporterOutput } from './utils'
+
+describe('tree reporter', () => {
+  it('reports a single test as a tree', async () => {
+    const { stdout } = await runVitest({
+      include: ['a.test.ts'],
+      root: 'fixtures/default',
+      reporters: ['tree'],
+    })
+
+    expect(trimReporterOutput(stdout)).toMatchInlineSnapshot(`
+      "❯ a.test.ts (16 tests | 1 failed | 3 skipped) [...]ms
+         ✓ a passed (6)
+           ✓ a1 test [...]ms
+           ✓ a2 test [...]ms
+           ✓ a3 test [...]ms
+           ✓ nested a (3)
+             ✓ nested a1 test [...]ms
+             ✓ nested a2 test [...]ms
+             ✓ nested a3 test [...]ms
+         ❯ a failed (7)
+           ✓ a failed 1 test [...]ms
+           ✓ a failed 2 test [...]ms
+           ✓ a failed 3 test [...]ms
+           × a failed test [...]ms
+           ✓ nested a failed (3)
+             ✓ nested a failed 1 test [...]ms
+             ✓ nested a failed 2 test [...]ms
+             ✓ nested a failed 3 test [...]ms
+         ✓ a skipped (3)
+           ↓ skipped with note [...]ms [reason]
+           ↓ condition [...]ms
+           ↓ condition with note [...]ms [note]"
+    `)
+  })
+
+  it('reports multiple tests when all passed as a tree', async () => {
```

Comment:
Took me a while to realize why some tests are reported as skipped, let's specify it in name too:

```suggestion
  it('reports multiple tests when all passed with testname filter as a tree', async () => {
```